ui-core/src/main/java/com/adyen/checkout/components/ui/view/AdyenComponentView.kt

```diff
@@ -85,14 +97,49 @@ class AdyenComponentView @JvmOverloads constructor(
 
         val componentView = viewType.viewProvider.getView(viewType, context, attrs, defStyleAttr)
         this.componentView = componentView
+        this.componentViewType = viewType
 
         val localizedContext = context.createLocalizedContext(componentParams.shopperLocale)
 
+        if (viewType is ButtonComponentViewType) {
+            binding.payButton.text = localizedContext.getString(viewType.buttonTextResId)
+        }
+
         val view = componentView.getView()
-        addView(view)
+        binding.frameLayoutComponentContainer.addView(view)
         view.updateLayoutParams { width = LayoutParams.MATCH_PARENT }
 
         componentView.initView(delegate, coroutineScope, localizedContext)
+
+        if (isConfirmationRequired) {
+            val uiStateDelegate = (delegate as? UIStateDelegate)
+            uiStateDelegate?.uiStateFlow?.onEach {
+                // setPaymentPendingInitialization has to be called on each event?
```

Comment:
Do we still need this comment? If so maybe add a TODO?

Reply:
Marking it as a TODO. Explained it as a response to @OscarSpruit 's comment.